.gitlab-ci.yml

```diff
@@ -1,29 +1,43 @@
-image: gcc:9.3
+default:
+  tags:
+    - docker
+  image: ${CI_REGISTRY}/${CI_PROJECT_PATH}/lpf-ubuntu-20.04-gcc
+
+variables:
+  IS_MAIN_PIPELINE: "no"
```

Comment:
The reason why I chose this pattern is because modifications to the rules are rare (I think they never happened since their introduction) and because at the same time there is only a few rule patterns that encompass multiple, related targets (e.g., build, install, unit, smoke, all of them for multiple configurations). In this context, I think it's cleaner and safer to "factor out" conditions into a variable that the jobs just read. If we make a change to the logic (e.g., trigger when the branch name also contains "release"), this likely involves entire groups of jobs, and can be applied only in one place.